code/game/objects/items/weapons/power_cells.dm

```diff
@@ -112,6 +112,27 @@
 	w_type = RECYK_BIOLOGICAL
 	minor_fault = 1
 
+/obj/item/weapon/cell/crepe
+	name = "power crepe"
+	desc = "Warning: May contain dairy products, 12,000kw of searing death, gluten."
+	origin_tech = Tc_POWERSTORAGE + "=3"
+	icon_state = "power_crepe"
+	maxcharge = 12000
+	charge = 12000
+	w_type = RECYK_BIOLOGICAL
+	minor_fault = 1
+
+/obj/item/weapon/cell/crepe/attack_self(var/mob/living/user)
+	user.visible_message("<span class = 'notice'>\The [user] takes a bite out of \the [src]</span>", "<span class = 'warning'>You take a bite out of \the [src]</span>")
+	spawn(rand(1,3) SECONDS)
+		var/power_to_use = min(charge, rand(800,1200))
+		playsound(loc, 'sound/effects/eleczap.ogg', 80, 1)
+		if(use(power_to_use))
+			user.adjustFireLoss(power_to_use/100) //So 8 to 12 damage
+			user.visible_message("<span class = 'notice'>\The [user] is electrocuted by \the [src]</span>", "<span class = 'warning'>You are [pick("frazzled","electrocuted","zapped")] by \the [src]!</span>")
+			user.set_light(2,2,"#ffff00")
```

Comment:
Well, how would you recommend doing it?

Most things that update lighting update it every so few ticks (The light virus effect, radglow, and I would assume fire), but there's no real way to revert back to that with this.

Two ways I can think of doing it. Store the light settings in temporary variables to be restored after the cooldown, or check if the person is alight already and if so don't light them up.